src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let correctSeparator = separator;
```

Comment:
you can simplify code and use only `separator` without `correctSeparator` - see other comments with explanations